Antoine/create inference image for listingclass

anotinelg opened the pull request:
**Description**

*JIRA ticket*: 

*What did you do?*

- forked from sagemaker-mxnet-inference-toolkit

- Add special docker directory for listing-class, where i created docker.cpu and docker.eia for the 1.4.1 and py3 framework/python.
- The docker.eia is a copy of the original one + some requirements needed for listingclass
- The docker.cpu is a copy of the original one + some changes in how we add dependency from sagemaker-inference-toolki (copied form the docker.eia files) + same requirements needed for listingclass. We have changed the way the sagemaker-inference-toolkit is imported because it fails with a version higher than 1.4.4
- We have added some script to build and push all the images.
- The tagging is done with the hash of the commit + possibility to add a postfix following semantic versioning.


*Is there any documentation?*

README

*How did you test your changes?*

- build and push all images

**Reviewer(s)** mouloud, Luca

**Deadline** 2020-06-23

**ETA** 20 minutes

**Expected Review**

*What do you want from the reviewers?*

- to understand the set of changes done on docker, look at the commit message https://github.com/letgoapp/sagemaker-mxnet-inference-toolkit/pull/1/commits/a91bda8e48eabdd40788ccef0f0aece94a4acd45  and https://github.com/letgoapp/sagemaker-mxnet-inference-toolkit/pull/1/commits/45b5ff0f6b3b18f4fb9223d907f0c880053b1dad
- validate decision on tagging (look the readme, and the scripts.  ./scripts/build_docker.py, scripts/push_docker.py). 
- test it (look Readme)

*Is there any specific part of the code you'd like to receive special attention?*

lcnletgo commented on scripts/build_docker.py:
```diff
@@ -0,0 +1,105 @@
+    os.chdir(prev_dir)
+
+
+def get_git_version():
```
I'd call this function "get_git_commit" (when reading the name I thought "why do we need to know the git version?"

anotinelg replied:
ok

lcnletgo commented on scripts/build_docker.py:
```diff
@@ -0,0 +1,105 @@
+            tag_git = get_git_version()
+            semversion = "" if args.semversion is None else f"-v{args.semversion}"
+            prev = '{}:{}-latest'.format(repo, tag)
+            dest = '{}:{}-{}{}'.format(repo, tag, tag_git,semversion)
```
I thought we were using only one of tag_git or semversion (or use two tags, one with git commit and one with the semversion)

anotinelg replied:
i use both.. yes it is a little bit different to what we spoke one week ago

lcnletgo commented on README.rst:
```diff
@@ -25,3 +25,44 @@ License
+python scripts/build_docker.py --docker-folder docker-listingclass \
+   --account 491446337008 \
+   --repo preprod-ds-listingclass-inference \
+   --version 1.4.1
```
can we call "--version" "--mxnet-version"?

lcnletgo replied:
also it's missing a `\`

anotinelg replied:
ok

lcnletgo commented on docker-listingclass/1.4.1/py3/Dockerfile.cpu:
```diff
@@ -0,0 +1,88 @@
+# TODO: Should we be installing OpenCV in our image like this? Is there another way we can fix this?
+RUN ln -s /dev/null /dev/raw1394
+
+ENV PYTHONDONTWRITEBYTECODE=1 \
```
oh nice, we should get these environment variables in the EC2 instances where we do the training then :)
(for example for LANG and LC_ALL we use "en.UTF-8", I am not sure if that could cause use problems or not at some point.

lcnletgo commented on docker-listingclass/1.4.1/py3/Dockerfile.cpu:
```diff
@@ -0,0 +1,88 @@
+FROM ubuntu:16.04
+
+LABEL maintainer="Amazon AI"
```
Maybe we can change this to letgo? or datascience@letgo.com?

lcnletgo commented on scripts/build_docker.py:
```diff
@@ -0,0 +1,105 @@
+# Copyright 2019-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
@anotinelg just to understand, this is a script from AWS?

anotinelg replied:
It is a script that i have modified

lcnletgo commented on scripts/push_docker.py:
```diff
@@ -0,0 +1,74 @@
+import logging
+import os
+
+DEFAULT_REGION = 'us-east-1'
```
shouldn't we pass this as a parameter?

anotinelg replied:
it is passed

lcnletgo commented on README.rst:
`````diff
@@ -25,3 +25,44 @@ License
+   --version 1.4.1
+   --semversion 0.1.0
+
+````
`````
there's an extra "`" which doesn't make it render it correctly from Github

lcnletgo replied:
I couldn't run the example, it complains about scripts/build_docker.py not existing, but the file does exist:

> ip-192-168-1-110:sagemaker-mxnet-inference-toolkit luca.cerone$ python scripts/build_docker.py --docker-folder docker-listingclass \
>    --account 491446337008 \
>    --repo preprod-ds-listingclass-inference \
>    --version 1.4.1 \
>    --semversion 0.1.0
  File "scripts/build_docker.py", line 43
    logging.warning(f"file {dockerfile_path} does not exist. Skipping")
                                                                     ^

> SyntaxError: invalid syntax
> ip-192-168-1-110:sagemaker-mxnet-inference-toolkit luca.cerone$ git branch
> \* antoine/create_inference_image_for_listingclass
>   master
> ip-192-168-1-110:sagemaker-mxnet-inference-toolkit luca.cerone$ 
> ip-192-168-1-110:sagemaker-mxnet-inference-toolkit luca.cerone$ ls scripts/build_docker.py 
> scripts/build_docker.py
> ip-192-168-1-110:sagemaker-mxnet-inference-toolkit luca.cerone$

lcnletgo commented on README.rst:
```diff
@@ -25,3 +25,44 @@ License
+   --account 491446337008 \
+   --repo preprod-ds-listingclass-inference \
+   --version 1.4.1
+   --semversion 0.1.0
```
missing a `\`

lcnletgo commented on README.rst:
````diff
@@ -25,3 +25,44 @@ License
+
+Example to build  all dockers existing in the docker-listingclass folder:
+```
+python scripts/build_docker.py --docker-folder docker-listingclass \
````
silly comment, on my Mac Python still defaults to Python 2.8 and the command was failing, maybe we can change `python`  to `python3` or mention explicitly in the Readme that the script works only with Python 3?

lcnletgo commented on README.rst:
````diff
@@ -25,3 +25,44 @@ License
+```
+python scripts/push_docker.py --docker-folder docker-listingclass \
+  --account 491446337008 --repo preprod-ds-listingclass-inference \
+   --version 1.4.1
````
`\` is missing

anotinelg commented:
> As discussed in person, maybe changing the error message,
> or clarifying a bit the expected behaviour would be good,
> but the code it is good to go (and for the versioning, we agreed that at the moment
> having the commit is more crucial than the version, and if needs change, we'll change the code in the future).

thanks @luca, i addressed your feedback in the following commit.
I will pushed the PR
